src/System.Management.Automation/engine/parser/Compiler.cs

```diff
@@ -982,19 +982,23 @@ private static Expression Coalesce(Expression left, Expression right)
             {
                 return left;
             }
-            else if (leftType == typeof(AutomationNull))
-            {
-                return right;
-            }
```

Comment:
Why removing this shortcut branch?

Reply:
See @SeeminglyScience's comment: https://github.com/PowerShell/PowerShell/issues/12655#issuecomment-628595935